### ISSUE TITLE
InputNumber can be cheated in order to enter symbols or letters

### ISSUE THREAD

**Issue by AlvaroTKT** [primefaces/primereact#6324]
### Discussed in https://github.com/orgs/primefaces/discussions/1597

<div type='discussions-op-text'>

<sup>Originally posted by **AlvaroTKT** April  4, 2024</sup>

Hello,

We've encountered a bug while implementing the InputNumber component from Prime React. It's possible to input non-numeric characters when typing double-tap characters (such as an accent "`").

**Steps to reproduce** : This issue can be reproduced using the library's component guide itself. If you focus on the component, type an accent, and then immediately type a letter, the component inserts that letter. At the event level, nothing is triggered (neither event nor e.value).

![image](https://github.com/primefaces/community/assets/124895195/81a89614-5f3a-4cae-bd80-f7c357927778)

Is there any workaround, or is it indeed confirmed as a bug?</div>

As @melloware points, It can be reproduced in primevue too.

![image](https://github.com/primefaces/community/assets/124895195/786ab2d5-bd3b-4607-aba4-1a8cff840129)

**Comment by FlipWarthog** [comment]
@AlvaroTKT Can you elaborate on the specific steps you're taking and/or take a screen capture? I assumed you were talking about [Alt code](https://en.wikipedia.org/wiki/Alt_code) but I don't think that's what you mean and in my testing they are properly handled.

**Comment by AlvaroTKT** [comment]
> @AlvaroTKT Can you elaborate on the specific steps you're taking and/or take a screen capture? I assumed you were talking about [Alt code](https://en.wikipedia.org/wiki/Alt_code) but I don't think that's what you mean and in my testing they are properly handled.


No, I wasn't referring to the Alt codes. I've made a gif showing the process in real time and asking the program to detail the keys. Actually, what I do is quickly press the "`" key and then a letter. I can write with preloaded text, and I can do it in an empty input field as you can see. 

![primeReact](https://github.com/primefaces/primereact/assets/124895195/939a5e45-8f6e-4e8a-90a7-87c8a907a7a3)

**Comment by melloware** [comment]
@AlvaroTKT can you try again with 10.6.3 released today?  I made some changes to InputNumber recently around keyboard support.  Let us know if it is still happening.

**Comment by AlvaroTKT** [comment]
> @AlvaroTKT can you try again with 10.6.3 released today? I made some changes to InputNumber recently around keyboard support. Let us know if it is still happening.

I tested it in my app and in documentation.

![image](https://github.com/primefaces/primereact/assets/124895195/d2727322-5209-48b6-bf7e-1e34ab2a3491)

It's still happening

**Comment by melloware** [comment]
OK thanks for reporting back.

**Comment by melloware** [comment]
Weird i am trying everything to reproduce and can't reproduce it.  @FlipWarthog and I are on US based keyboards do you think that makes a difference?

**Comment by AlvaroTKT** [comment]
Here is the screenshot with the results:

![image](https://github.com/primefaces/primereact/assets/124895195/7ca33e74-02a9-468c-89a5-07ebfe879397)

I also add a gif with a suite of cases of tick key triggering. tick with consonant, tick with tick and tick with vowel for adding more info

![kEventViewer](https://github.com/primefaces/primereact/assets/124895195/5397bf0d-2e73-4581-8559-dbe23950467c)

**Comment by melloware** [comment]
Nice @FlipWarthog and I will study this.  The code seems to be "Dead".   I will look at how this can happen.

**Comment by FlipWarthog** [comment]
@melloware https://en.wikipedia.org/wiki/Dead_key

I think we can use virtual keyboards in Windows set to Spanish or another language that has a Dead key for replicating this.

**Comment by melloware** [comment]
Yeah from what i was reading it said change your Windows Keyboard to Spanish or Italian?

**Comment by FlipWarthog** [comment]
Yup, from the Windows settings you can add a language. Once you have more than one installed it'll add a lang switcher to the taskbar in the bottom right. 

![image](https://github.com/primefaces/primereact/assets/83613837/480ab77f-5bee-40fe-b3a1-3cbffc45f9cf)

This updates your keyboard layout and also updates the on-screen keyboard to match (so you can actually see what they keys are)

![image](https://github.com/primefaces/primereact/assets/83613837/be533204-c846-41dc-a24f-83f870444327)

I was able to replicate the issue following these steps.

**Comment by melloware** [comment]
OK PR submitted.

**Comment by AlvaroTKT** [comment]
Thanks for the hard work ^^

**Comment by dodvarko** [comment]
Hi guys, this is still the case with some symbols on Czech keyboards, such as `'` and `ˇ`. attaching screenshots from the keyboard event viewer:

<img width="1131" alt="Image" src="https://github.com/user-attachments/assets/948998c5-0233-4d50-a68f-0b9454ad834e" />
<img width="1088" alt="Image" src="https://github.com/user-attachments/assets/6e9a93e5-237d-49a9-b9eb-6a93288c36c3" />

 I have no way of reopening the ticket, let me know whether i should create a new one. Thanks

**Comment by melloware** [comment]
@dodvarko please open a new ticket and reference this one and please add all your screenshots!
